### PR TITLE
4.0-rc1

### PR DESCRIPTION
**Pull request by sosodev**
* remove makefile and clarify build instructions in readme
* add DAEMON_OLD_VERSION health check (addresses #194)
* remove unused health check code
* ~~filter duplicate OSD nodes to prevent errors in osd_down collection~~ #99 should actually be fixed already
* add TLS support (addresses #205)

### CODE REVIEW

**Comment by sosodev** [comment]
It be good to get a fix in for #102 too but that's going to require significantly more refactoring. I think what we have here is already a major improvement over the current release.

File: main.go [hunk 1]
```diff
@@ -157,8 +160,15 @@ func main() {
 		logrus.WithError(err).Fatal("error creating listener")
 	}
 
-	err = http.Serve(emfileAwareTcpListener{ln.(*net.TCPListener), logger}, nil)
-	if err != nil {
-		logrus.WithError(err).Fatal("error serving requests")
+	if len(*tlsCertPath) != 0 && len(*tlsKeyPath) != 0 {
+		err = http.ServeTLS(emfileAwareTcpListener{ln.(*net.TCPListener), logger}, nil, *tlsCertPath, *tlsKeyPath)
```
Comment (baergj) [inline main.go:164]: Should we be using something like https://github.com/dyson/certman here? (That one seems old - just using it as an example.)

Comment (sosodev) [reply main.go:164]: Hmm yeah that could be a good addition.

File: ceph/osd.go [hunk 2]
```diff
@@ -905,6 +902,24 @@ func (o *OSDCollector) collectOSDTreeDown(ch chan<- prometheus.Metric) error {
...
+				return true
+			}
+		}
+
+		return false
+	}
+
+	for _, downOsdNode := range downItems {
+		if !filteredContainsNode(downOsdNode) {
+			filteredDownItems = append(filteredDownItems, downOsdNode)
```
[the remark this answers is not in the document]
Comment (sosodev) [reply ceph/osd.go:919]: Oops good catch. Removing it anyway since you pointed out that it may not be necessary. :)

File: main.go (hunk 1 above)
Comment (sosodev) [reply main.go:164]: Is loading the cert on every request an acceptable solution? Seems like it adds fairly minimal overhead and the implementation is much simpler than using file watchers

Comment (baergj) [reply main.go]: Huh, interesting. That's probably fine for an exporter, but I'm far from an expert in this area.

Comment (sosodev) [reply main.go]: https://github.com/digitalocean/ceph_exporter/pull/210/commits/ef01452f1cd4b7c3cf6affc8b94ff3ac974dbb4b what that looks like

**Pull request merged**